src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +435,123 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue) 
+{
+    this.CustomHttpHeaders = new Dictionary <string, string> ();
+
+    if (String.IsNullOrWhiteSpace(stringValue)) {
+        return;
+    }
+
+    if (ValidateParantheses(stringValue) && ValidateFirstLast(stringValue) && ValidateDelimeters(stringValue)) 
+    {
+        string header = stringValue.TrimEnd('}', ' ');
+        string[] elements = header.Split('}');
+        foreach (var element in elements)
+        {
+            string _element = element.TrimStart(',', '{', ' ');
+            string[] array = _element.Split(':');
+            CustomHttpHeaders.Add(array[0], array[1]);
+        }
+    }
+    else 
+    {
+        throw new ArgumentException("Please check your custom http headers string format! You have set the custom http header but not use the format '{headerName:value},{headerName:value}'");
+    }
+
+}
+
+private static bool ValidateParantheses(string _input)
+{
+    Dictionary<char, char> paraPairs = new Dictionary<char, char>() {
+        { '{', '}' }
+    };
+
+    Stack<char> stack = new Stack<char>();
+
+    foreach (char i in _input)
+    {
+        if(stack.Count > 1)
+        {
+            return false;
+        }
+        if (paraPairs.Keys.Contains(i))
+        {
+            stack.Push(i);
+        }                    
+        else if(paraPairs.Values.Contains(i)) 
+        {
+            if(stack.Count > 0 && paraPairs[stack.Peek()] == i)
+            {
+                stack.Pop();
+            }
+            else
+            {
+                return false;
+            }
+        }
+    }
+    return stack.Count() == 0 ? true : false;
+}
+
+private static bool ValidateFirstLast(string _input)
+{
+    string input = _input.Trim();
+    char firstElement = input[0];
+    char lastElement = input[input.Length - 1];
+
+    if(firstElement == '{' && lastElement == '}')
+    {
+        return true;
+    }
+    return false;
+}
+
+private static int GetMatchedDelimeters(string _input)
+{
+    // Remove white spaces
+    string headers = Regex.Replace(_input, @"\s", "");
+
+    // Regex pattern },{
+    string delimeterPattern = @"[\}][,][\{]";
+    Regex rgx = new Regex(delimeterPattern);
+    // Get all matches  
+    MatchCollection matchedHeaders = rgx.Matches(headers);

```

Comment:
check the security to use Regx?

Reply:
This regex pattern is evaluated in a linear manner instead of backtracking. Therefore it isn't prone to ReDOS (Regex Denial Of Service)